go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -389,10 +401,19 @@ func (c *Controller) generateFRRConfigurations(ra *ratypes.RouteAdvertisements)
 		selectedNetworks.vrfs = append(selectedNetworks.vrfs, vrf)
 		selectedNetworks.networkVRFs[vrf] = networkName
 		selectedNetworks.networkTopology[networkName] = network.TopologyType()
-		// TODO check overlaps?
 		for _, cidr := range network.Subnets() {
 			subnet := cidr.CIDR.String()
 			len := uint32(cidr.HostSubnetLength)
+
+			// Check for conflicts within the same RouteAdvertisements targeting default VRF
+			if ra.Spec.TargetVRF == "" {
+				for _, existingSubnet := range selectedNetworks.subnets {
+					if subnetsOverlap(subnet, existingSubnet) {
+						return nil, nil, fmt.Errorf("%w: subnet %s from network %q conflicts with subnet %s from another selected network within the same RouteAdvertisements %q targeting default VRF",
```

Comment:
I think we can improve the error message; let's try to be less verbose, and more to the point: 
"failed exposing network %q: overlapping CIDR detected: %q" 

What do you think ?

Reply:
I ended up using a more verbose error message, cause I think it matters to the user to know where the overlapping occured (inside the RA? on a different RA?).
Let me know what you think of what I did and if you still think it's too verbose then we could further trim it.